src/rulesets-new/property-rules.ts

```diff
@@ -0,0 +1,393 @@
+  matches: (_, ruleContext) =>
+    ruleContext.operation.change !== "added" &&
+    !isBreakingChangeAllowed(ruleContext.custom.changeVersion.stability),
+  rule: (requestAssertions) => {
+    requestAssertions.property.removed("not be removed", (property) => {
+      throw new RuleError({
+        message: `Expected property ${property.value.key} to not be removed`,
+      });
+    });
+  },
+});
+
+const responsePropertyRemoval = new ResponseBodyRule({
+  name: "response property removal",
+  docsLink: links.versioning.breakingChanges,
+  matches: (_, ruleContext) =>
+    ruleContext.operation.change !== "added" &&
+    !isBreakingChangeAllowed(ruleContext.custom.changeVersion.stability),
+  rule: (responseAssertions) => {
+    responseAssertions.property.removed("not be removed", (property) => {
+      throw new RuleError({
+        message: `Expected property ${property.value.key} to not be removed`,
+      });
+    });
+  },
+});
+
+const requiredRequestProperties = new RequestRule({
+  name: "prevent adding a required request property",
+  docsLink: links.versioning.breakingChanges,
+  matches: (_, ruleContext) =>
+    ruleContext.operation.change !== "added" &&
+    !isBreakingChangeAllowed(ruleContext.custom.changeVersion.stability),
+  rule: (requestAssertions) => {
+    requestAssertions.property.added(
+      "not add required request property",
+      (property) => {
+        if (property.value.required) {
+          throw new RuleError({
+            message:
+              "cannot add a required request property to an existing operation",
+          });
+        }
+      },
+    );
+
+    requestAssertions.property.changed(
+      "not make an optional request property required",
+      (before, after) => {
+        if (!before.value.required && after.value.required) {
+          throw new RuleError({
+            message: "cannot make a request property required",
+          });
+        }
+      },
+    );
+  },
+});
+
+function withinAttributes(context) {
+  if (!("jsonSchemaTrail" in context)) return false;
+  const { jsonSchemaTrail } = context;
+  // We don't want to check [data, attributes] or [data, items, attributes]
+  // so we return false for anything that isn't nested deeper.
+  if (
+    (jsonSchemaTrail[0] === "data" &&
+      jsonSchemaTrail[1] === "attributes" &&
+      jsonSchemaTrail.length > 2) ||
+    (jsonSchemaTrail[0] === "data" &&
+      jsonSchemaTrail[0] === "items" &&
+      jsonSchemaTrail[2] === "attributes" &&
+      jsonSchemaTrail.length > 3)
+  )
+    return true;
+  return false;
+}
+
+const enumOrExample = new RequestRule({
+  name: "request property enum or example",
+  docsLink: links.standards.formats,
+  rule: (requestAssertions) => {
+    requestAssertions.property.added("have enum or example", (property) => {
+      const type = property.value.flatSchema.type;
+      if (!withinAttributes(property.location)) return;
+      if (type === "object" || type === "boolean") return;
```

Comment:
The reason we can't add this in the matches block, is because this runs on _every_ property. The nuance is kind of hard to communicate, but on matches, you are matching the top level item (e.g. RequestRule will match a request), where as requestAssertions.properties, are property assertions for the request